src/base/system_parameter.c

```diff
@@ -7374,6 +7511,27 @@ sysprm_reload_and_init (const char *db_name, const char *conf_file)
   return sysprm_load_and_init_internal (db_name, conf_file, true, SYSPRM_LOAD_ALL);
 }
 
+
+
+static bool
+prm_section_cmp (const char *key1, const char *key2, int key_len, bool ignore_case)
+{
+  const char *s1 = strchr (key1, ':');
+  int len = (int) ((s1) ? CAST_STRLEN (s1 - key1) : strlen (key1));
```

Comment:
```suggestion
  int len = (s1) ? CAST_STRLEN (s1 - key1) : strlen (key1);
```
이미 캐스팅되어 있어서 추가적인 캐스팅이 필요하지 않을을 거 같습니다.

Reply:
strlen()의 결과도 casting 대상이 되어야 합니다.  size_t  --> int

Reply:
해당 strlen은  다음과 같이 정의되어 있으며, SUPPRESS_STRLEN_WARNING 매크로는 porting.h 에서 항상 정의되어지고 있습니다. 저 매크로를 정의한 이유는 모르겠지만 warning을 제거하기 위해 작성된 거 같습니다.
```c
// system_parameter.c

#include "porting.h"
...

#if defined (SUPPRESS_STRLEN_WARNING)
#define strlen(s1)  ((int) strlen(s1))
#endif /* defined (SUPPRESS_STRLEN_WARNING) */
```

Reply:
아 그렇게 설정이 되어 있군요, 
감사합니다.